sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-part-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -67,6 +73,359 @@ describe(
+      expect(
+        mockedJob.containLogMessages([
+          `Creating notifications for disbursement id: ${disbursementScheduleId} for student and ministry.`,
+          `Completed creating notifications for disbursement id: ${disbursementScheduleId} for student and ministry.`,
+        ]),
+      ).toBe(true);
+      const studentNotificationCount = await db.notification.count({
+        where: {
+          notificationMessage: {
+            id: NotificationMessageType.StudentNotificationDisbursementBlocked,
+          },
+          metadata: {
+            disbursementId:
+              application.currentAssessment.disbursementSchedules[0].id,
+          },
+          user: { id: application.student.user.id },
+          creator: { id: systemUsersService.systemUser.id },
+        },
+      });
+      expect(studentNotificationCount).toBe(1);
+      const ministryNotificationCount = await db.notification.count({
+        where: {
+          notificationMessage: {
+            id: NotificationMessageType.MinistryNotificationDisbursementBlocked,
+          },
+          metadata: {
+            disbursementId:
+              application.currentAssessment.disbursementSchedules[0].id,
+          },
+          user: { id: systemUsersService.systemUser.id },
+          creator: { id: systemUsersService.systemUser.id },
+        },
+      });
+      expect(ministryNotificationCount).toBe(1);
+    });
+
+    it("Should not create a notification for the student for a disbursement when there are already 3 notifications created.", async () => {
+      // Arrange
+      // Student with invalid SIN to block the disbursement.
+      const student = await saveFakeStudent(db.dataSource, undefined, {
+        sinValidationInitialValue: {
+          isValidSIN: false,
+        },
+      });
+      // Valid MSFAA Number.
+      const msfaaNumber = await db.msfaaNumber.save(
+        createFakeMSFAANumber(
+          { student },
+          {
+            msfaaState: MSFAAStates.Signed,
+            msfaaInitialValues: {
+              offeringIntensity: OfferingIntensity.partTime,
+            },
+          },
+        ),
+      );
+      // Student application.
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { student, msfaaNumber },
+        {
+          offeringIntensity: OfferingIntensity.partTime,
+          applicationStatus: ApplicationStatus.Completed,
+          firstDisbursementInitialValues: {
+            coeStatus: COEStatus.completed,
+          },
+        },
+      );
+      // Create pre-existing 3 notifications for the above created disbursement.
+      let notificationCounter = 3;
+      while (notificationCounter > 0 && notificationCounter--) {
+        await saveFakeNotification(
+          db.dataSource,
+          {
+            user: student.user,
+            creator: systemUsersService.systemUser,
+            notificationMessage: {
```

Comment:
while creating fake notification data, please add a date_sent so that we can check for new notifications created with date_sent is NULL

Reply:
I was just thinking if it is required to check for `date_sent` here. Any suggestions: @andrewsignori-aot @andrepestana-aot @guru-aot @lewischen-aot

Reply:
Checking for the proper log message can also ensure that the new notification was not generated.
Checking the date_sent would be a valid way also to test.